props/proptypes.md

```diff
@@ -0,0 +1,226 @@
+
+It would be nice if the React library could enforce the presence or data type of our props...
+
+## PropTypes
+
+**PropTypes** is a React feature that allows us to explicitly list the props a component takes and their types.
+
+When we use PropTypes effectively, PropTypes will:
+
+- Provide type-checking and validation of props
+- Create implied self-documentation of our components
+
+Both of these benefit our developer workflow and communication!
+
+## PropType Syntax
+
+We define PropTypes per component.
+
+For each component, we need to do the following:
+
+1. Import `PropTypes` into our component file
+1. Attach a `propTypes` member to our component function
+1. Assign the `propTypes` member to an object, where each key-value pair is a `prop` name and a _validator_
+
+Here, we are adding PropTypes to a component `ComponentA`.
+
+In this example, these PropTypes state that there are two expected `props` for `ComponentA`:
+
+- `title`, which is a string, and required
+- `operate`, which is a function
+
+<!-- prettier-ignore-start -->
+```js
+import PropTypes from 'prop-types';
+
+const ComponentA = (props) => {
+    return (
+        <span>{props.title}</span>
+    )
+}
+
+ComponentA.propTypes = {
+    title: PropTypes.string.isRequired,
+    operate: PropTypes.func
+};
+
+export default ComponentA;
+```
+<!-- prettier-ignore-end -->
+
+| <div style="min-width:200px;"> Piece of Code </div> | Notes                                                                                                                                          |
+| --------------------------------------------------- | ---------------------------------------------------------------------------------------------------------------------------------------------- |
+| `import PropTypes from 'prop-types';`               | We need to import `PropTypes` properly before we use it.                                                                                       |
+| `ComponentA.propTypes`                              | After and outside the `ComponentA` function definition, we will set a value to the `propTypes` member of `ComponentA`.                         |
+| `= { ... };`                                        | The value of `ComponentA.propTypes` should be an object.                                                                                       |
+| `title:`                                            | **Replace this** with the exact name of a `prop` we want to validate.                                                                          |
+| `PropTypes.string.isRequired`                       | **Replace this** with a PropType validator. Details below. This example validator checks if the `title` prop is a string, and if it's present. |
+| `operate: PropTypes.func`                           | Each props-validator pair is comma-separated. This props-validator pair states that the value of the prop `operate` should be a function.      |
```

Comment:
prefer whether when it's the state itself that's important
reinforce that the comma separation is normal JS object syntax
fully explain the second validator (that the lack of isRequired makes it optional)
```suggestion
| `PropTypes.string.isRequired`                       | **Replace this** with a PropType validator. Details below. This example validator checks whether the prop is a string, and whether it's present. |
| `operate: PropTypes.func`                           | Each props-validator pair is comma-separated, following regular JavaScript object syntax. This props-validator pair states that the value of the prop `operate` should be a function, but that it is optional (not required).      |
```